homeassistant/components/nws/weather.py

```diff
@@ -165,85 +73,73 @@ def convert_condition(time, weather):
         raise PlatformNotReady
 
-    _LOGGER.debug("Station list: %s", nws.stations)
-    _LOGGER.debug(
-        "Initialized for coordinates %s, %s -> station %s",
-        latitude,
-        longitude,
-        nws.station,
-    )
+    await nws_data.async_update()
 
-    async_add_entities([NWSWeather(nws, mode, hass.config.units, config)], True)
+    async_add_entities(
+        [
+            NWSWeather(nws_data, "daynight", hass.config.units),
+            NWSWeather(nws_data, "hourly", hass.config.units),
+        ],
+        False,
+    )
 
 
 class NWSWeather(WeatherEntity):
     """Representation of a weather condition."""
 
-    def __init__(self, nws, mode, units, config):
+    def __init__(self, nws, mode, units):
         """Initialise the platform with a data instance and station name."""
         self.nws = nws
-        self.station_name = config.get(CONF_NAME, self.nws.station)
+        self.station = nws.station
+        self.latitude = nws.latitude
+        self.longitude = nws.longitude
+
         self.is_metric = units.is_metric
         self.mode = mode
 
         self.observation = None
         self._forecast = None
 
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    async def async_update(self):
-        """Update Condition."""
-        _LOGGER.debug("Updating station observations %s", self.nws.station)
-        try:
-            await self.nws.update_observation()
-        except ERRORS as status:
-            _LOGGER.error(
-                "Error updating observation from station %s: %s",
-                self.nws.station,
-                status,
-            )
-        else:
-            self.observation = self.nws.observation
-        _LOGGER.debug("Observation: %s", self.observation)
-        _LOGGER.debug("Updating forecast")
-        try:
-            if self.mode == "daynight":
-                await self.nws.update_forecast()
-            else:
-                await self.nws.update_forecast_hourly()
-        except ERRORS as status:
-            _LOGGER.error(
-                "Error updating forecast from station %s: %s", self.nws.station, status
+    async def async_added_to_hass(self) -> None:
+        """Set up a listener and load data."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                signal_unique_id(self.latitude, self.longitude),
+                self._update_callback,
             )
-            return
+        )
+        self._update_callback()
+
+    @callback
+    def _update_callback(self) -> None:
+        """Load data from integration."""
+        self.observation = self.nws.observation
         if self.mode == "daynight":
             self._forecast = self.nws.forecast
         else:
             self._forecast = self.nws.forecast_hourly
-        _LOGGER.debug("Forecast: %s", self._forecast)
-        _LOGGER.debug("Finished updating")
+
+        self.async_schedule_update_ha_state()
```

Comment:
This should be `self.async_write_ha_state` since we don't need to fetch and update data.